zict/zip.py

```diff
@@ -24,53 +29,61 @@ class Zip(MutableMapping):
     >>> z.flush()  # flush and write metadata to disk  # doctest: +SKIP
     """
 
-    def __init__(self, filename, mode="a"):
+    filename: str
+    mode: Literal["r", "w", "x", "a", "closed"]
```

Comment:
For stuff like this I prefer to do some type arithmetic, which makes keeping things in sync easier if it needs to change, as well as makes it easier to see the differences, IMO:

```python
MODES = Literal["r", "w", "x", "a"]

...
    mode: MODES | Literal["closed"]
...
    def __init__(self, filename: str, mode: MODES = "a"):
```